### PR TITLE
Set a token default and support private repos

### PR DESCRIPTION
**Pull request by peter-evans**
The aim of this PR is to improve the interface and allow the action to work for private repositories.

- Adds a `token` input that defaults to the default `GITHUB_TOKEN` in the repository.
- Supersedes https://github.com/jaxxstorm/action-install-gh-release/pull/11

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -21,12 +21,7 @@ async function run() {
     try {
 
         // set up auth/environment
-        const token = process.env['GITHUB_TOKEN']
-        if (!token) {
-            throw new Error(
-                `No GitHub token found`
-            )
-        }
+        const token = process.env['GITHUB_TOKEN'] || core.getInput("token")
```
Comment (peter-evans) [inline src/main.ts:24]: For backwards compatibility the environment variable is used if available. If not it falls back to the `token` input.

File: action.yml [hunk 2]
```diff
@@ -2,6 +2,9 @@ name: "Install a binary from GitHub releases"
 description: "Install binaries from GitHub releases"
 author: "Lee Briggs"
 inputs:
+  token:
+    description: 'GITHUB_TOKEN or a `repo` scoped Personal Access Token (PAT)'
+    default: ${{ github.token }}
```
Comment (peter-evans) [inline action.yml:7]: Defaults to `GITHUB_TOKEN` so users don't need to specify a token in workflows unless they want to access a private repository.

**Comment by jaxxstorm** [comment]
thanks for the awesome contribution!

**Comment by peter-evans** [comment]
Thank you for maintaining this great action!

**Pull request merged**